Chapter-4-applying-tactical-domain-driven-design/README.adoc

```diff
@@ -40,29 +77,238 @@ IMPORTANT: It makes no sense to consider the Domain Model in typical CRUD module
+
+Let's proceed with and anlysis and take a look on __Annex__. Annexes are a part of the __Binding Contract__ <<aggregate,aggregate>>. They have to be uniquely identified and encapsulate business logic. We will model it as an <<entity, entity>>.
+
+__Binding Contract__ has signature property which has business logic and is no requirement to be uniquely identified. Signature can be compared by its properties. We want to use the ubiquitous language, so we have chosen <<value-object, value object>> as building block to model this concept.
+
+Every time we attach annex to the __Binding Contract__, we want to notify other parts of the system about this event. This is a perfect use case for <<domain-events,Domain Events>>.
+
+[[building_blocks]]
+=== Building Blocks:
+
+[[entity]]
+==== Entity
+
+An **Entity** is representation of business concept that has its own identity. It is defined by its attributes, behavior, and identity. Entities are used to model objects that have a lifecycle and are mutable.
+
+==== Main characteristics of entities:
+
+- They have a unique identity
+- They represent a business concept
+- They have behavior (methods)
+- They have state (properties)
+- They can be changed over time
+- They encapsulate business logic
+- They can raise domain events after creating or state change
+- They can be internal part of an aggregate root
+- They can be becoming aggregate root when needed of protecting invariants
+- In one bounded context concept can be modeled as a value object and in another as an entity
+
+[source,csharp]
+----
+public sealed class Annex : Entity
+{
+    public AnnexId Id { get; init; } // Unique Entity Id
+    public BindingContractId BindingContractId { get; init; }
+    public DateTimeOffset ValidFrom { get; init; } // State
+
+    // EF needs this constructor to create non-primitive types
+    private Annex() { }
+
+    private Annex(BindingContractId bindingContractId, DateTimeOffset validFrom)
+    {
+        Id = AnnexId.Create();
+        BindingContractId = bindingContractId;
+        ValidFrom = validFrom;
+
+        var @event = AnnexAttachedToBindingContractEvent.Raise(Id, BindingContractId, ValidFrom); // Raise domain event
+        RecordEvent(@event); 
+    }
+
+    internal static Annex Attach(BindingContractId bindingContractId, DateTimeOffset validFrom) =>
+        new(bindingContractId, validFrom); // Behavior method
+}
+----
+
+NOTE: You probably have heard about Anemic Domain Model. This is known as anti-pattern. It is an entity that has only properties and no behavior. It is acceptable when you have simple CRUD operations. In complex process, we recommend to encapsulate behavior (methods) in the domain entity insted having this logic in the service layer.
```

Comment:
I would drop this line. To use behavioral entities you do not need to use tactical DDD, it can be either active record or a plain, behavioral entity